app/models/room.rb

```diff
@@ -82,12 +92,28 @@ def set_meeting_id
 
   # Autocreate all meeting options using the default values
   def create_meeting_options
-    MeetingOption.all.find_each do |option|
-      RoomMeetingOption.create(room: self, meeting_option: option, value: option.default_value)
+    access_codes_names = %w[glViewerAccessCode ModeratorAccessCode]
+    access_configs = MeetingOption.access_codes_configs(provider: 'greenlight').to_h
+
+    MeetingOption.find_each do |option|
+      value = if access_codes_names.include?(option.name) && access_configs[option.name] == 'true'
```

Comment:
`access_codes_names.include ?(...)` check was added as a safe guard in case the `MeetingOption.access_codes_configs` returns some irrelevant settings because of an intended update to its implementation ensuring the determinism of this action,